kumascript/src/info.js

```diff
@@ -162,15 +162,55 @@ const info = {
       return this.getPageByURL(document);
     }
 
-    const { locale, slug, title, summary, tags } = document.metadata;
+    const { locale, slug, title, tags } = document.metadata;
     return {
       url: document.url,
       locale,
       slug,
       title,
-      summary,
       tags: tags || [],
       translations: [], //TODO Object.freeze(buildTranslationObjects(data)),
+      get summary() {
+        // Back in the old Kuma days we used to store the summary as another piece
+        // of metadata on each document. It was always available, with any kumascript
+        // macros rendered out.
+        // In Yari, this is not possible. We don't duplicate the summary in every
+        // document. Instead, we extract it from the document when we build it.
+        // So, to avoid the whole chicken-and-egg problem, instead, we're going to
+        // use regular expressions to try to extract it on-the-fly, from the
+        // raw HTML.
+        // Note, we can't use Cheerio here because the `document.rawHTML` is
+        // actually not valid HTML, hence the desperate fall back on regex.
+        // A lot of times, you'll actually find that the first paragraph isn't
+        // a <p> tag. But often, in those cases it'll have that `seoSummary`
+        // <span> tag. Like this for example:
+        //
+        //   <div><span class="seoSummary">The <code>window.stop()</code> ...
+        //
+        // So that's why we always start by looking for that tag first.
+        try {
+          const seoSummaryMatch = document.rawHTML.match(
+            /<span class="seoSummary">(.*?)<\/span>/
+          );
+          if (seoSummaryMatch) {
+            return seoSummaryMatch[1];
+          }
+
+          const matches = document.rawHTML.matchAll(/<p[^>]*>(.*?)<\/p>/gs);
```

Comment:
I noticed that this matches paragraphs that are really warning blocks, like the first paragraph of `/en-US/docs/Mozilla/Add-ons/WebExtensions/API/proxy`

<img width="894" alt="image" src="https://user-images.githubusercontent.com/3743693/96649583-47b8bf00-12e6-11eb-9920-0edb36353142.png">


It'd be nice to filter those paragraphs out, so we could add a simple "negative look-behind assertion" to do that:

```suggestion
          const matches = document.rawHTML.matchAll(
            /(?<!<div class="blockIndicator warning">\s*)<p[^>]*>(.*?)<\/p>/gs
          );
```

which I've tested and it works. You can see the difference for the `proxy` link on http://localhost:5000/en-US/docs/Mozilla/Add-ons/WebExtensions/API/#JavaScript_API_listing:

**Before:**
<img width="550" alt="image" src="https://user-images.githubusercontent.com/3743693/96650139-299f8e80-12e7-11eb-964f-86b1d90c76ca.png">

**After:**
<img width="541" alt="image" src="https://user-images.githubusercontent.com/3743693/96649939-de857b80-12e6-11eb-85f0-c6a62a3f2341.png">